src/OrchardCore.Modules/OrchardCore.Apis.GraphQL/SchemaMiddleware.cs

```diff
@@ -0,0 +1,30 @@
+using System.Net;
+using System.Threading.Tasks;
+using GraphQL.Utilities;
+using Microsoft.AspNetCore.Http;
+
+namespace OrchardCore.Apis.GraphQL
+{
+    public class SchemaMiddleware
+    {
+        private readonly RequestDelegate _next;
+
+        public SchemaMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task Invoke(HttpContext context, ISchemaFactory schemaService)
+        {
+            var schema = await schemaService.GetSchemaAsync();
```

Comment:
we should probably check permissions .. dont wanna leak the schema just because graphql is turned on ...